docs/architecture/cloud-native/grpc.md

```diff
@@ -0,0 +1,113 @@
+
+The book, [gRPC for WCF Developers](https://docs.microsoft.com/dotnet/architecture/grpc-for-wcf-developers/), available from the Microsoft Architecture site, provides in-depth coverage of gRPC and Protocol Buffers.
+
+## gRPC support in .NET
+
+gRPC is integrated into .NET Core 3.0 SDK or later. The following tools support it:
+
+- Visual Studio 2019, version 16.3 or later, with the web development workload installed.
+- Visual Studio Code
+- the dotnet CLI
+
+The SDK includes tooling for endpoint routing, built-in IoC, and logging. The open-source Kestrel web server supports HTTP/2 connections. Figure 4-20 shows a Visual Studio 2019 template that scaffolds a skeleton project for a gRPC service. Note how .NET Core fully supports Windows, Linux, and macOS.
+
+![gRPC Support in Visual Studio 2019](./media/visual-studio-2019-grpc-template.png)
+
+**Figure 4-20**. gRPC support in Visual Studio 2019
+  
+Figure 4-21 shows the skeleton gRPC service generated from the built-in scaffolding included in Visual Studio 2019.  
+
+![gRPC project in Visual Studio 2019](./media/grpc-project.png  )
+
+**Figure 4-21**. gRPC project in Visual Studio 2019
+
+In the previous figure, note the proto description file and service code. As you'll see shortly, Visual Studio generates additional configuration in both the Startup class and underlying project file.
+
+## gRPC usage
+
+Favor gRPC for the following scenarios:
+
+- Synchronous backend microservice-to-microservice communication where an immediate response is required to continue processing.
+- Polyglot environments that need to support mixed programming platforms.
+- Low latency and high throughput communication where performance is critical.
+- Point-to-point real-time communication - gRPC can push messages in real time without polling and has excellent support for bi-directional streaming.
+- Network constrained environments – binary gRPC messages are always smaller than an equivalent text-based JSON message.
+
+At the time, of this writing, gRPC is primarily used in backend services. Most modern browsers can't provide the level of HTTP/2 control required to support a front-end gRPC client. That said, there's an [early initiative](https://devblogs.microsoft.com/aspnet/grpc-web-experiment/) that enables gRPC communication from browser-based apps built with JavaScript or Blazor WebAssembly technologies. The  [gRPC-Web for .NET](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md) enables an ASP.NET Core gRPC app to support gRPC features in browser apps:
+
+- Strongly-typed code-generated clients
+- Compact Protobuf messages
+- Server streaming
+
+## gRPC implementation
+
+The microservice reference architecture, [eShop on Containers](https://github.com/dotnet-architecture/eShopOnContainers), from Microsoft, shows how to implement gRPC services in .NET Core applications. Figure 4-22 presents the back-end architecture.
+
+![Backend architecture for eShop on Containers](./media/eshopWithAggregators.png)
```

Comment:
Can you fix the image name here? Standardize to use with hyphens. I think you already have an image with the name `eshop-grpc-wide.png` which can be used here? If yes, drop the `eshopWithAggregators.png` and update the reference.

Reply:
Great catch. Done